src/compiler/utilities.ts

```diff
@@ -2744,6 +2744,20 @@ namespace ts {
     }
 
     /**
+     * Gets the effective return type annotation of a signature. If the node was parsed in a
+     * JavaScript file, gets the return type annotation from JSDoc.
+     */
+    export function getEffectiveTypeParameterDeclarations(node: DeclarationWithTypeParameters): TypeParameterDeclaration[] {

```

Comment:
Is there normally a perf impact for moving functions to utilities? I didn't expect anything. But I can check to see whether anything shows up.